sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala

```diff
@@ -82,15 +82,15 @@ object StreamingQueryListener {
    * @param id A unique query id that persists across restarts. See `StreamingQuery.id()`.
    * @param runId A query id that is unique for every start/restart. See `StreamingQuery.runId()`.
    * @param name User-specified name of the query, null if not specified.
-   * @param submissionTime The timestamp to start a query.
+   * @param timestamp The timestamp to start a query.
    * @since 2.1.0
    */
   @Evolving
   class QueryStartedEvent private[sql](
       val id: UUID,
       val runId: UUID,
       val name: String,
-      val submissionTime: Long) extends Event
+      val timestamp: String) extends Event
```

Comment:
Change this to `string` type to be consistent with timestamps in `StreamingQueryProgress`.

Reply:
`timestamp` is not used without parsing and the parsed result is the same as [startTimestamp](https://github.com/apache/spark/pull/27732/files#diff-6532dd3b63bdab0364fbcf2303e290e4R310). So `formatTimestamp(startTimestamp)` seems a little bit meaningless.
I understand it's for consistency but how about rather changing the type of `StreamingQueryProgress#timestamp` to `Long` for consistency?
Almost all the use of `StreamingQueryProgress#timestamp` is with parsing so could we reduce the parsing code?

Reply:
Yep, totally agreed that `Long` is better for coding. However, `StreamingQueryProgress#timestamp` was designed to be human readable. In addition, its type cannot be changed now because that's a public API. Since the user has already had codes to parse `StreamingQueryProgress.timestamp`, it should be fine to add a field with the same format.

Reply:
Ah, It's a public API. O.K, I understand that it's reasonable.